### DISCUSSION TITLE
请教：什么是透明代理ipv6

### DISCUSSION THREAD

**Discussion by yunyuyuan** [vernesong/OpenClash#2983] [Q&A]
## 前提
openclash运行在路由器openwrt上，有一个连接路由器的linux设备A。
在不勾选openclash的`代理ipv6`情况下，路由器和A都能正常使用ipv6，用手机流量(ipv6)访问A的http服务也没问题。
此时ddns查询`curl 6.ipw.cn`返回的是A的ipv6，而`curl ipv6.google.com`是无法访问的。
## 如何透明代理ipv6
如果我勾上了`代理ipv6`，`curl ipv6.google.com`可以访问，但是ddns查询`curl 6.ipw.cn`返回的是路由器的ipv6地址而不是A的ipv6。用手机流量(ipv6)仍然可以访问A，可以认为openclash代理了ipv6的出站流量，没有处理ipv6的入站。

我不是很了解ipv6网络协议，在网上没看懂**透明代理ipv6**的意思。我理解的**透明**就是A不知道自己被代理了，例如：A可以正常访问墙外服务`ipv6.google.com`，而且`ipv6.google.com`的server发现请求ip是A的ipv6，而不是路由器或梯子的ipv6。请问我理解错了吗？这是可以实现的吗？

据我猜测，ipv4是无法实现这个的，`google.com`的server收到的请求ip肯定是梯子的ipv4

**Comment by macaujack** [comment]
同问，我开「代理 ipv6」时， `curl 6.ipw.cn` 也是返回路由器的 ipv6 地址，访问 https://test-ipv6.com 或者 https://ipv6.test-ipv6.com 时也显示我的地址是路由器的 ipv6 地址，想知道原因是什么，所谓的「透明代理」不应该对请求发起端和接收端都无感知吗？

顺便一提，访问 https://www.ip5.me/ipv6 时，能显示我的 ipv6 地址是设备 A 的。
